src/config/wmodules-azure.c

```diff
@@ -480,6 +468,22 @@ int wm_azure_storage_read(const OS_XML *xml, XML_NODE nodes, wm_azure_storage_t
                 OS_ClearNode(children);
             }
 
+        // To avoid the defects reported by scan-build in the issue: https://github.com/wazuh/wazuh/issues/11568
+        // It's necessary to check that nodes[i]->content is valid
+        } else if (nodes[i]->content) {
+            if (!strcmp(nodes[i]->element, XML_ACCOUNT_NAME)) {
+                if (*nodes[i]->content != '\0')
+                    os_strdup(nodes[i]->content, storage->account_name);
+            } else if (!strcmp(nodes[i]->element, XML_ACCOUNT_KEY)) {
+                if (nodes[i]->content && *nodes[i]->content != '\0')
+                    os_strdup(nodes[i]->content, storage->account_key);
+            } else if (!strcmp(nodes[i]->element, XML_AUTH_PATH)) {
+                if (nodes[i]->content && *nodes[i]->content != '\0')
+                    os_strdup(nodes[i]->content, storage->auth_path);
+            } else if (!strcmp(nodes[i]->element, XML_TAG)) {
+                if (nodes[i]->content && *nodes[i]->content != '\0')
+                    os_strdup(nodes[i]->content, storage->tag);
+            }
```

Comment:
```suggestion
        } else if (NULL == nodes[i]->content && '\0' != *nodes[i]->content) {
            if (!strcmp(nodes[i]->element, XML_ACCOUNT_NAME)) {
                os_strdup(nodes[i]->content, storage->account_name);
            } else if (!strcmp(nodes[i]->element, XML_ACCOUNT_KEY)) {
                os_strdup(nodes[i]->content, storage->account_key);
            } else if (!strcmp(nodes[i]->element, XML_AUTH_PATH)) {
                os_strdup(nodes[i]->content, storage->auth_path);
            } else if (!strcmp(nodes[i]->element, XML_TAG)) {
                os_strdup(nodes[i]->content, storage->tag);
            } else {
                merror(XML_INVELEM, nodes[i]->element);
                return OS_INVALID;
            }
```
And remove the next else block
https://github.com/wazuh/wazuh/pull/11581/files#diff-58cfe0f7636d9a05b1650f0bc7ee18c1c0789f00026195b1146d4d7b27b5bddcL484-L487